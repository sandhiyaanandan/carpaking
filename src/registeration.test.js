import React from 'react';
import { shallow, mount } from 'enzyme';
import Registeration from './registeration';
import {findByTestAttr} from './../jestutils';
import toJson from 'enzyme-to-json';

const setUp =()=>{
    const component = shallow(<Registeration />);
    return component;
};

describe('Registeration', ()=>{
  let wrapper;
    beforeEach(() =>{
        wrapper = setUp();
    });

it('should render the Registeration Component correctly', ()=>{   
      expect(toJson(wrapper)).toMatchSnapshot();
    });

it('should render Registeration component correctly', ()=>{
  let comp = findByTestAttr(wrapper, 'registeration');
  expect(comp.length).toBe(1);
});

// make our assertion and what we expect to happen 
 it('should render RegisterForm without throwing an error', () => {
   expect(wrapper.find('form#regForm').exists()).toBe(true);
 })

describe(' Register Name', () => { 
 it('should respond to change event and change the state of the Login name', () => {
   wrapper.find('[name="userName"]').simulate('change', {target: {name:"userName", value: 'Uuser'}});
   expect(wrapper.state('userName')).toEqual('Uuser');
  })
});

describe('Register Password', () => {
 it('should respond to change event and change the state of the Password', () => {
   wrapper.find('[name="password"]').simulate('change', {target: {value: 'admin'}});
   expect(wrapper.state('password')).toEqual('admin');
  })
});
describe('Register Confirm Password', () => {
 it('should respond to change event and change the state of the Confirm Password', () => {
   wrapper.find('[name="confirmPwd"]').simulate('change', {target: {value: 'admin'}});
   expect(wrapper.state('password')).toEqual('admin');
  })
});
describe('Register CarName', () => {
 it('should respond to change event and change the state of the Carname', () => {
   wrapper.find('[name="carName"]').simulate('change', {target: {value: 'Toyota'}});
   expect(wrapper.state('carName')).toEqual('Toyota');
  })
});

describe('Register CarNumber', () => {
 it('should respond to change event and change the state of the Carnumber', () => {
   wrapper.find('[name="carNumber"]').simulate('change', {target: {value: 'TN78JK8976'}});
   expect(wrapper.state('carNumber')).toEqual('TN78JK8976');
  })
});

});