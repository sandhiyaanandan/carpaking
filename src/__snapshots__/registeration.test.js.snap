// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registeration should render the Registeration Component correctly 1`] = `
<styled.div>
  <div
    data-test="registeration"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  >
    <div
      className="loginBox"
      style={
        Object {
          "backgroundColor": "blue",
          "border": "10px solid blue",
        }
      }
    >
      <h2
        className="pageHeader"
      >
        Register
      </h2>
      <form
        id="regForm"
        style={
          Object {
            "marginTop": "10%",
          }
        }
      >
        <div>
          <styled.input
            name="userName"
            onChange={[Function]}
            placeholder="Enter User Name"
            type="text"
          />
          <styled.input
            name="password"
            onChange={[Function]}
            placeholder="Enter Password"
            type="password"
          />
          <styled.input
            name="confirmPwd"
            onChange={[Function]}
            placeholder="Enter Confirm Password"
            type="password"
          />
          <styled.input
            name="carName"
            onChange={[Function]}
            placeholder="Enter Carname"
            type="text"
          />
          <styled.input
            name="carNumber"
            onChange={[Function]}
            placeholder="Enter CarNumber"
            type="text"
          />
        </div>
        <styled.input
          className="submitButton"
          onClick={[Function]}
          style={
            Object {
              "marginTop": "20%",
            }
          }
          type="button"
          value="Register"
        />
      </form>
    </div>
  </div>
</styled.div>
`;
